apps/api/src/controllers/applicationController.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response } from 'express';
+import applicationService from '@/service/application-service.js';
+import { getDbInstance } from '@/db/index.js';
+import { OrderBy, type ApplicationService } from '@/service/types.js';
+import { ApplicationStateValues } from '@pcgl-daco/data-model/src/types.js';
+
+/**
+ * Handles the creation of a new application.
+ * Endpoint: POST /applications
+ */
+export const createApplicationController = async (req: Request, res: Response) => {
+  const { user_id } = req.body;
+
+  if (!user_id) {
+    return res.status(400).json({ success: false, message: 'User ID is required' });
+  }
+
+  try {
+    const database = getDbInstance();
+	const service: ApplicationService = applicationService(database);
+    const result = await service.createApplication({ user_id });
+    return res.status(result.success ? 200 : 400).json(result);
+  } catch (error) {
+    console.error('Error creating application:', error);
+    return res.status(500).json({ success: false, message: 'Internal server error' });
+  }
+};
+
+/**
+ * Handles editing an existing application.
+ * Endpoint: PUT /applications/:id
+ */
+export const editApplicationController = async (req: Request, res: Response) => {
+  const { id } = req.params;
+  const update = req.body;
+
+  if (!id || !update) {
+    return res.status(400).json({ success: false, message: 'Application ID and update data are required' });
+  }
+
+  try {
+    const database = getDbInstance();
+	const service: ApplicationService = applicationService(database);
+    const result = await service.editApplication({ id: Number(id), update });
+    return res.status(result.success ? 200 : 400).json(result);
+  } catch (error) {
+    console.error('Error editing application:', error);
+    return res.status(500).json({ success: false, message: 'Internal server error' });
+  }
+};
+
+/**
+ * Fetches all applications based on filters.
+ * Endpoint: GET /applications
+ */
+export const getAllApplicationsController = async (req: Request, res: Response) => {
+  const { userId, state, sort, page, pageSize } = req.query;
+
+  try {
+    const database = getDbInstance();
+	const service: ApplicationService = applicationService(database);
+    const result = await service.listApplications({ 
+        user_id: userId as string | undefined,
+      state: state as ApplicationStateValues | undefined,
+      sort: sort ? [sort as OrderBy<"user_id" | "id" | "created_at" | "updated_at" | "state" | "approved_at" | "expires_at" | "contents">] : undefined,
+      page: page ? Number(page) : undefined,
+      pageSize: pageSize ? Number(pageSize) : undefined,
+     });
```

Comment:
There is a lot to change with `listApplications` here.
Compare with https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/src/api/application-api.ts#L86
We have a type definition for `ApplicationListRequest` so that you don't need to define the types for each field like this.
I also generally recommend putting this logic into variable definitions, not in the function arguments, for debugging clarity.

We avoid using `as` in Typescript projects (unless really needed) because it allows you to 'lie' to Typescript. I can refer you to a draft style guide we have for TS projects that covers a few additional team conventions to keep in mind.